src/pipelines/sentence_embeddings/pipeline.rs

```diff
@@ -0,0 +1,370 @@
+            (DistilBert(_), ConfigOption::DistilBert(conf)) => conf.n_layers as usize,
+            (DistilBert(_), _) => unreachable!(),
+            (Roberta(_), ConfigOption::Bert(conf)) => conf.num_hidden_layers as usize,
+            (Roberta(_), _) => unreachable!(),
+            (Albert(_), ConfigOption::Albert(conf)) => conf.num_hidden_layers as usize,
+            (Albert(_), _) => unreachable!(),
+            (T5(_), ConfigOption::T5(conf)) => conf.num_layers as usize,
+            (T5(_), _) => unreachable!(),
+        }
+    }
+
+    fn nb_heads(&self) -> usize {
+        use SentenceEmbeddingsOption::*;
+        match (&self.transformer, &self.transformer_config) {
+            (Bert(_), ConfigOption::Bert(conf)) => conf.num_attention_heads as usize,
+            (Bert(_), _) => unreachable!(),
+            (DistilBert(_), ConfigOption::DistilBert(conf)) => conf.n_heads as usize,
+            (DistilBert(_), _) => unreachable!(),
+            (Roberta(_), ConfigOption::Roberta(conf)) => conf.num_attention_heads as usize,
+            (Roberta(_), _) => unreachable!(),
+            (Albert(_), ConfigOption::Albert(conf)) => conf.num_attention_heads as usize,
+            (Albert(_), _) => unreachable!(),
+            (T5(_), ConfigOption::T5(conf)) => conf.num_heads as usize,
+            (T5(_), _) => unreachable!(),
+        }
+    }
+
+    pub fn encode_with_attention<S>(
+        &self,
+        inputs: &[S],
+    ) -> Result<(Vec<Embedding>, Vec<AttentionOutput>), RustBertError>
+    where
+        S: AsRef<str> + Sync,
+    {
+        let SentenceEmbeddingsModelOuput {
+            embeddings,
+            all_attentions,
+        } = self.forward(inputs)?;
+
+        let embeddings = Vec::from(embeddings);
+        let all_attentions = all_attentions.ok_or_else(|| {
+            RustBertError::InvalidConfigurationError("No attention outputted".into())
+        })?;
+
+        let attention_outputs = (0..inputs.len() as i64)
+            .map(|i| {
+                let mut attention_output = AttentionOutput::with_capacity(self.nb_layers());
+                for layer in all_attentions.iter() {
+                    let mut attention_layer = AttentionLayer::with_capacity(self.nb_heads());
+                    for head in 0..self.nb_heads() {
+                        let attention_slice = layer
+                            .slice(0, i, i + 1, 1)
+                            .slice(1, head as i64, head as i64 + 1, 1)
+                            .squeeze();
+                        let attention_head = AttentionHead::from(attention_slice);
+                        attention_layer.push(attention_head);
+                    }
+                    attention_output.push(attention_layer);
+                }
+                attention_output
+            })
+            .collect::<Vec<AttentionOutput>>();
+
+        Ok((embeddings, attention_outputs))
+    }
+}
+
+pub struct SentenceEmbeddingsTokenizerOuput {
```

Comment:
This probably does not need to be public

Reply:
Sure, it needs to be public only if `tokenize(...)` remains public too.